charts/sentinel-archiver/templates/statefulset.yaml

```diff
@@ -0,0 +1,325 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}-archiver
+  labels:
+    {{- include "sentinel-archiver.allLabels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  serviceName: {{ .Release.Name }}-archiver-service
+  selector:
+    matchLabels:
+      {{- include "sentinel-archiver.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "sentinel-archiver.allLabels" . | nindent 8 }}
+    spec:
+      # we're going to need root for a little while until
+      # we can package some helper binaries into the image
+      #securityContext:
+        #fsGroup: 532
+        #runAsNonRoot: true
+        #runAsUser: 532
+        #runAsGroup: 532
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      imagePullSecrets:
+      - name: regcred
+      initContainers:
+      - name: initfs
+        image: {{ required "(root).image.repo expected" .Values.lily.image.repo }}:{{ default (printf "v%s" .Chart.AppVersion) .Values.lily.image.tag }}
+        imagePullPolicy: {{ .Values.lily.image.pullPolicy }}
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            mkdir -p /var/lib/lily/keystore
+            # generate 6-char random uid to be used in job report names
+            if [ ! -f "/var/lib/lily/uid" ]; then
+              tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w ${1:-6} | head -n 1 > /var/lib/lily/uid
+            fi
+        volumeMounts:
+        - name: repo-volume
+          mountPath: /var/lib/lily
+        - name: waitjob-script-volume
+          mountPath: /var/lib/lily/waitjob.sh
+          subPath: waitjob.sh
+        resources:
+          requests:
+            cpu: "1000m"
+            memory: "512Mi"
+          limits:
+            cpu: "1000m"
+            memory: "512Mi"
+      {{- if .Values.lily.importSnapshot.enabled }}
+      - name: chain-import
+        image: {{ required "(root).image.repo expected" .Values.lily.image.repo }}:{{ default (printf "v%s" .Chart.AppVersion) .Values.lily.image.tag }}
+        imagePullPolicy: {{ .Values.lily.image.pullPolicy }}
+        command: ["/bin/sh", "-c"]
+        args:
+          - |{{- include "sentinel-archiver.chainImportArgs" . | indent 10 }}
+        env:
+        - name: GOLOG_LOG_FMT
+          value: {{ .Values.logFormat | default "json" | quote }}
+        - name: LILY_REPO
+          value: "/var/lib/lily"
```

Comment:
Should we include lily envvars here? There may be some that would apply to import.